Python 3.10 Static Method Changes

The following test fails on Python 3.10:

```
======================================================================
FAIL: test_text (pdoc.test.CliTest) (package='example_pkg')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-python3.10-pdoc3-0.10.0.drv-0/pdoc3-0.10.0/pdoc/test/__init__.py", line 372, in test_text
    self.assertNotIn(pattern, out)
AssertionError: 'non_callable_routine' unexpectedly found in "Module example_pkg ...
```

I think this is because, according [to documentation](https://docs.python.org/3/library/functions.html#staticmethod):

> Changed in version 3.10: Static methods now inherit the method attributes (__module__, __name__, __qualname__, __doc__ and __annotations__), have a new __wrapped__ attribute, and are now callable as regular functions.

I apologize I don't know enough how to fix this but would be happy to contribute what I can.

--- a/pkgs/development/python-modules/pdoc3/default.nix
+++ b/pkgs/development/python-modules/pdoc3/default.nix
@@ -17,7 +17,7 @@ buildPythonPackage rec {
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "5f22e7bcb969006738e1aa4219c75a32f34c2d62d46dc9d2fb2d3e0b0287e4b7";
+    hash = "sha256-XyLnvLlpAGc44apCGcdaMvNMLWLUbcnS+y0+CwKH5Lc=";
   };
 
   patches = [
@@ -25,7 +25,14 @@ buildPythonPackage rec {
       # test_Class_params fails in 0.10.0
       # https://github.com/pdoc3/pdoc/issues/355
       url = "https://github.com/pdoc3/pdoc/commit/4aa70de2221a34a3003a7e5f52a9b91965f0e359.patch";
-      sha256 = "07sbf7bh09vgd5z1lbay604rz7rhg88414whs6iy60wwbvkz5c2v";
+      hash = "sha256-W7Dy516cA+Oj0ZCTQBB6MJ+fCTBeLRp+aW8nANdxSx8=";
+    })
+    # https://github.com/pdoc3/pdoc/issues/400
+    (fetchpatch {
+      name = "fix-test-for-python310.patch";
+      url = "https://github.com/pdoc3/pdoc/commit/80af5d40d3ca39e2701c44941c1003ae6a280799.patch";
+      hash = "sha256-69Cn+BY7feisSHugONIF/PRgEDEfnvnS/RBHWv1P8/w=";
+      excludes = [".github/workflows/ci.yml"];
     })
   ];
 
@@ -40,7 +47,6 @@ buildPythonPackage rec {
   ];
 
   meta = with lib; {
-    broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
     description = "Auto-generate API documentation for Python projects.";
     homepage = "https://pdoc3.github.io/pdoc/";
     license = with licenses; [ agpl3Plus ];

